### PR TITLE
Documentation fixes

### PR DESCRIPTION
**Pull request by riedgar-ms**
Various updates to the documentation:

- Rename `contents.html` to `index.html`
- Move migration guides under the user guide
- Link to migration guide from quickstart
- Other small edits

### CODE REVIEW

File: docs/about/index.rst
Comment (MiroDudik) [inline docs/about/index.rst:31]: i don't think we need this paragraph here now that we have a note in quickstart and user guide.

**Review by MiroDudik — APPROVED** [review]

**Comment by MiroDudik** [comment]
Please also check: https://github.com/fairlearn/fairlearn/pull/612/files#r517659422

File: docs/conf.py
Comment (romanlutz) [inline docs/conf.py:88]: What changed? This was intentional to my knowledge since index is our custom landing page. 

[There's a PR out #618 that seems related to me, just like the related issue.]

File: docs/user_guide/migrating_versions/index.rst
Comment (romanlutz) [inline docs/user_guide/migrating_versions/index.rst:3]: nit: lowercase p and v

File: docs/conf.py
Comment (MiroDudik) [reply docs/conf.py:88]: I didn't catch the conversation on #570 (also tagging @leonarduschen). Basically, I chatted with @riedgar-ms and we thought that it's better if the root of the version `<version>` is: `https://fairlearn.github.io/<version>/index.html`. 

(Committing to `https://fairlearn.github.io/<version>/contents.html` as the root seems very unusual.)

Also take a look at the version selectors on:
https://fairlearn.github.io/master/contents.html
https://fairlearn.github.io/v0.4.6/index.html

Versus what you get with the current PR:
https://1871-133444044-gh.circle-artifacts.com/0/docs/_build/html/master/index.html
https://1871-133444044-gh.circle-artifacts.com/0/docs/_build/html/v0.4.6/index.html

Do you see any issues with this update?

**Re. what changed**: My guess is that previously we didn't have a versioned documentation, so there was a conflict between the "landing page" https://fairlearn.github.io/ and the "root of the documentation". That conflict has now disappeared. The final tweak that we will need to do is to redirect https://fairlearn.github.io/ to point at `https://fairlearn.github.io/<most-recent-version>/index.html`--but I think we should hold off with that until the landing page redesign (and that redesigned landing page then becomes the new "version root"). At that point, we will follow a similar pattern as https://scikit-learn.org/

Comment (riedgar-ms) [reply docs/conf.py:88]: I'm pretty sure @MiroDudik  correctly outlined the reason why we had `contents.html` before - I think you did that bit @romanlutz ? Now that we have versioned documentation, and each version is its own mini-website with the  custom-designed landing page separate and non-versioned, so we can go back to `index.html`.

As for changing the default links from the landing page, I do have an item on the release checklist to update the links to point at the latest release. In due course, we'll need to write a script to do that automatically as part of the release process.

File: docs/user_guide/migrating_versions/index.rst
Comment (riedgar-ms) [reply docs/user_guide/migrating_versions/index.rst:3]: We haven't been entirely consistent on that sort of point, but I can change this one.

File: docs/conf.py
Comment (leonarduschen) [reply docs/conf.py:88]: As far as my understanding goes, setting `master_doc` to `contents` is just a (somewhat hacky) way to make a custom landing page. The side effect of this is - clicking the logo will then leads to `contents` instead of `index` because `pydata-sphinx-theme` sets it to `master_doc` by default.

#618 is supposed to fix this issue by bumping `pydata-sphinx-theme`. The new feature would allow customizing where the logo leads when clicked (internal HTML page or external link). That said, if `index` is indeed the desired logo link, then this PR should fix #570

Comment (MiroDudik) [reply docs/conf.py:88]: @leonarduschen: Thanks for clarifying!

I think that a more basic issue that is being addressed by this PR is that we had no `index.html` at the root level.

Comment (romanlutz) [reply docs/conf.py:88]: Thanks for the description, @MiroDudik ! I had indeed not realized that this adds a separate one per version. In fact, that solves more issues than I had initially realized! Wonderful :-)

**Review by romanlutz — APPROVED** [review]

**Pull request merged**
